### ISSUE TITLE
Optimize unrealcv performance

### ISSUE THREAD

**Issue by qiuwch** [unrealcv/unrealcv#19]
Currently, the speed of unrealcv is roughly 10 - 20fps on my workstation, it is enough for generating image and video dataset, but not enough for interesting tasks such as reinforcement learning. Optimizing its speed will be the next focus.

**Comment by qiuwch** [comment]
Another thing needs to be done is better object mask format. Current format (png) is hard to parse and slow to use. Also I need to implement a direct communication between python and UE4, instead of saving to the disk first.

**Comment by xd009642** [comment]
I was thinking of potentially tackling this myself for my own usage. Wasn't sure if merging back in would be of interest? I was thinking of using protobuf as a serialisation/deserialisation method.

**Comment by qiuwch** [comment]
@xd009642, thanks. I am very interested in a protobuf implementation. 
Previously I spent some time reading these two repos.
https://github.com/code4game/libprotobuf
https://github.com/PaddleCreekGames/Proto3RPC_UE4

These repos should be a good starting point. Challenges for using protobuf in UE4 is cross-platform support and distributing the protobuf code.
